### DISCUSSION TITLE
Where should future development happen?

### DISCUSSION THREAD

**Discussion by kylegentle** [Byron/google-apis-rs#350] [Q&A]
Hey @Byron! Thanks again for your help enabling custom connectors for `hyper` clients in #338.

I'm interested in contributing some further work to these crates, specifically:
1. Get the `admin` APIs to generate successfully
2. Provide a way to do batch requests, as described in the [GDrive API docs](https://developers.google.com/drive/api/guides/performance#batch_requests)

I noticed that there's an [alternate implementation](https://github.com/google-apis-rs/generator) with some nice design goals, but it seems like most recent activity has been limited to the original implementation, and this repo is still where all the crates are generated & published from.

Personally, having more experience with Python than Rust, I'm probably better equipped to contribute here. I'd be happy to make some incremental refactors aimed at cleaning up the Python + Mako codebase. But I don't want to further diverge the project if it's still planned to move forward with the newer implementation.

Where's the best place to direct my effort?

**Comment by Byron** [comment]
> Thanks again for your help enabling custom connectors for hyper clients in https://github.com/Byron/google-apis-rs/pull/338.

Thanks for making the contribution :)!

> I noticed that there's an [alternate implementation](https://github.com/google-apis-rs/generator) with some nice design goals, but it seems like most recent activity has been limited to the original implementation, and this repo is still where all the crates are generated & published from.

Indeed, development there seems to have stalled despite already being somewhat usable. Originally I created that group and was an owner, but have recently removed myself from it due to the inactivity and me not working on it anyway. This might be useful context for…

> Where's the best place to direct my effort?

The only place where I have control is this repository, and despite me wanting to see it being replaced by an alternative implementation, it really seems there's life in the old dog yet. Thus I welcome you to make this project maintainable once again in any way you see fit. It would be great to see the makefile/buildsystem repaired and CI working again, for instance 😅. 

Eventually, and if you are interested, I will happily welcome you as co-maintainer of these crates in which you seem to have great interest, possibly culminating in ownership transfer if you want. After all, I am not using these crates myself and maybe everyone would be served better if they were in the hands of someone who has greater need for them.

In any case, please be welcome to contribute here and take these crates into the future - I think they are valuable for the Rust ecosystem and deserve to be kept in better shape than they are in now.

**Comment by kylegentle** [comment]
I've been making some progress at [kylegentle:python_fixes](https://github.com/kylegentle/google-apis-rs/tree/python_fixes).

My intent for this first chunk of work (and likely the next few) is to make incremental refactoring & type annotation changes without modifying the behavior of the codegen. I've verified this by re-generating all APIs at the base and tip of my local branch; the only changes are to the "this file was generated automatically from x" lines, as I've moved & renamed some things.

If you like what you see so far, I'm happy to submit a PR, or copy the branch here and continue working. Just let me know what kind of workflow you'd prefer.

I'm certainly open to the idea of co-maintaining the crates in the future; let's see where this improvement effort goes! :smile:

**Reply by Byron** [reply]
I like the approach of using the generated files as test to assure the refactoring doesn't change behavior, it makes perfect sense :)! Besides, everything you do will be an improvement I have no doubt, and for any feedback an early PR would be a way for me to provide it.

It's good for me to know the current plans/goals as well like you just shared so I can act as QA, this is really how I see my role along with some advisory sprinkled in.

Take 'em into the future :).
